snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedReceivedSnapshot.java

```diff
@@ -169,12 +171,16 @@ private boolean isSnapshotIdInvalid(final String snapshotId) {
   }
 
   private boolean writeReceivedSnapshotChunk(
-      final SnapshotChunk snapshotChunk, final Path snapshotFile) throws IOException {
-    Files.write(
-        snapshotFile,
-        snapshotChunk.getContent(),
-        StandardOpenOption.CREATE_NEW,
-        StandardOpenOption.WRITE);
+      final SnapshotChunk snapshotChunk, final Path snapshotFile) {
+    try (var file = new RandomAccessFile(snapshotFile.toFile(), "ws")) {
+      final FileChannel channel = file.getChannel();
+      final ByteBuffer buffer = ByteBuffer.wrap(snapshotChunk.getContent());
+      channel.write(buffer);
+    } catch (IOException e) {
```

Comment:
Why catch the error now? Actually, I wonder why bother with returning a boolean and let the error bubble up?

Reply:
Because the apply method returns true or false for sucess/failure. Bubbling up the error wouldn't meet that interface and it would fail the actor

Reply:
But previously we did let it bubble, which is why I was confused by the change. Can you elaborate on the differences?

Reply:
What I said was wrong, my bad. The interface does expect an IOException and its consumers handle it.  Although I think letting an exception bubble up would fail the actor and complete the future exceptionally so that's even more confusing. Seems like there's 3 ways this interface can inform a consumer of a failure: completing the future with false, completing the future exceptionally and throwing an exception. I can refactor this, should I do it in this PR or separately?

Reply:
Let's do it in a separate PR so we can merge this now. What's your plan for refactoring it?

Reply:
I'll have to look closely but off the top of my head: remove the IOException from the interface, catch exceptions in the implementation and complete the future exceptionally. Then change `Future<Boolean>` to `Future<Void>` and throw some SnapshotFailureException instead of completing the future with false. Then the consumers wouldn't need to catch exceptions, check for exceptions in the future and for a future completed with "false". They could just check if the future is completed with an exception.